src/coreclr/jit/fgprofile.cpp

```diff
@@ -1448,9 +1448,10 @@ class ClassProbeVisitor final : public GenTreeVisitor<ClassProbeVisitor<TFunctor
     Compiler::fgWalkResult PreOrderVisit(GenTree** use, GenTree* user)
     {
         GenTree* const node = *use;
-        if (node->IsCall())
+        if (node->IsCall() && (node->AsCall()->gtCallMoreFlags & GTF_CALL_M_CLASS_PROBE))
```

Comment:
Can we just check `call->gtClassProfileCandidateInfo != nullptr` instead of adding another flag?

Reply:
That sounds reasonable to me too.

Reply:
@jakobbotsch @AndyAyersMS done